bokehjs/src/lib/models/axes/axis.ts

```diff
@@ -154,6 +155,23 @@ export abstract class AxisView extends GuideRendererView {
     return super.is_renderable && range.is_valid && cross_range.is_valid && range.span > 0 && cross_range.span > 0
   }
 
+  protected abstract _hit_value(sx: number, sy: number): any | null
```

Comment:
This should be  `number | Factor | null` given how it's used later. Or it could be `unknown | null`. Note that `unknown | null` and `any | null` reduce to `unknown` or `any` respectively, but it's good for the sake of documentation to write those `null` types down anyway.

Reply:
In derived classes you can either use the generic type or specific types like `Factor | null` in the categorical case.

Reply:
[931b309](https://github.com/bokeh/bokeh/pull/14050/commits/931b3099e22f90acc337e5264c9b2bbb48750b9e)